app/logic/questionnaire_expander.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class QuestionnaireExpander
+  class << self
+    def expand_content(content, response)
+      raise unless content.is_a? Array
+      content.reduce([]) do |total, question|
+        total.concat(question[:foreach].present? ? process_foreach(question, response) : [question])
+      end
+    end
+
+    private
+
+    def process_foreach(question, response)
+      res = []
+      case question[:foreach]
+      when :student
+        response.person.my_students.each_with_index do |person, idx|
+          res << replace_question(question.deep_dup, person, idx)
+        end
+      else
+        raise "Foreach option #{question[:foreach]} not found"
+      end
+      res
+    end
+
+    # TODO: We'd probably want to change idx to an actual identifier
+    def replace_question(question, person, idx)
+      question = replace_student_names(question, idx)
+      question = replace_student_zijn_haar(question, idx)
+      question = replace_student_hij_zij(question, idx)
+      question = replace_student_hem_haar(question, idx)
+      question = replace_id(question, person)
+      question
+    end
```

Comment:
Kun je hier niet gewoon dezelfde variabelenamen voor gebruiken? dwz `naam_student` ipv `naam_studenten` etc? En dat je dan alleen in de vragen met een `:foreach` de `naam_student` dingen gaat vervangen door `naam_student_1` oid. Dat zou volgens mij gewoon goed moeten gaan.

Dan ziet het er al een stuk netter uit, en dan is het ook minder verwarrend in de notatie.